spec/PasswordPolicy.spec.js

```diff
@@ -1011,4 +1011,304 @@ describe("Password Policy: ", () => {
     });
   });
 
+  it('should fail if passwordPolicy.passwordHistory is not a number', done => {
+    reconfigureServer({
+      appName: 'passwordPolicy',
+      passwordPolicy: {
+        passwordHistory: "not a number"
+      },
+      publicServerURL: "http://localhost:8378/1"
+    }).then(() => {
+      fail('passwordPolicy.passwordHistory "not a number" test failed');
+      done();
+    }).catch(err => {
+      expect(err).toEqual('passwordPolicy.passwordHistory must be an integer ranging 0 - 20');
+      done();
+    });
+  });
+
+  it('should fail if passwordPolicy.passwordHistory is a negative number', done => {
+    reconfigureServer({
+      appName: 'passwordPolicy',
+      passwordPolicy: {
+        passwordHistory: -10
+      },
+      publicServerURL: "http://localhost:8378/1"
+    }).then(() => {
+      fail('passwordPolicy.passwordHistory negative number test failed');
+      done();
+    }).catch(err => {
+      expect(err).toEqual('passwordPolicy.passwordHistory must be an integer ranging 0 - 20');
+      done();
+    });
+  });
+
+  it('should fail if passwordPolicy.passwordHistory is greater than 20', done => {
+    reconfigureServer({
+      appName: 'passwordPolicy',
+      passwordPolicy: {
+        passwordHistory: 21
+      },
+      publicServerURL: "http://localhost:8378/1"
+    }).then(() => {
+      fail('passwordPolicy.passwordHistory negative number test failed');
+      done();
+    }).catch(err => {
+      expect(err).toEqual('passwordPolicy.passwordHistory must be an integer ranging 0 - 20');
+      done();
+    });
+  });
+
+  it('should fail if the new password is same as the last password', done => {
+    const user = new Parse.User();
+    const emailAdapter = {
```

Comment:
@bhaskaryasa 
Do we really need an email adapter here? I think this test case can be simplified to something  like: 

```
  it('should fail if the new password is same as the last password', done => {
    reconfigureServer({
      appName: 'passwordPolicy',
      passwordPolicy: {
        passwordHistory: 1
      },
    }).then(() => {
      Parse.User.signUp('user1', 'password1').then(user1 => {
        user1.setPassword('password1');
        return user1.save();
      }).then(() => {
        // cannot set the same password twice
        fail();
      }).catch(err => {
        expect(err.message).to.equal('New password should not be the same as last 1 passwords.');
        expect(err.code).to.equal(142); // 142 might have to be replaced by Parse.Error.xxx
        done();
      });
    });
  });
```

WDYT?